code/datums/components/overlay_lighting.dm

```diff
@@ -245,8 +245,9 @@
 		return
 	if(new_holder != parent && new_holder != parent_attached_to)
 		RegisterSignal(new_holder, COMSIG_PARENT_QDELETING, .proc/on_holder_qdel)
-		RegisterSignal(new_holder, COMSIG_MOVABLE_MOVED, .proc/on_holder_moved)
 		RegisterSignal(new_holder, COMSIG_LIGHT_EATER_QUEUE, .proc/on_light_eater)
+		if(overlay_lighting_flags & LIGHTING_ON)
+			RegisterSignal(new_holder, COMSIG_MOVABLE_MOVED, .proc/on_holder_moved)
```

Comment:
parent must always have its movement registered, but the holder (which is usually the human mob, since pda's carry flashlights) should only have their movement registered if the light is actually on and needs updates